docs/guides/choosing-a-subject-type.md

```diff
@@ -0,0 +1,89 @@
+# Choosing a Subject Type
+
+Within a Permissions System, a [CheckPermission] call is always made between an object representing the *resource* and an object representing a *subject*: The API call returns whether the *subject* has the specified permission on the *resource*.
+
+[CheckPermission]: https://buf.build/authzed/api/docs/main/authzed.api.v1#CheckPermission
+
+The choice of how to represent subjects can have large downstream effects on the design of a permissions system.
+
+## Representing users as subjects
+
+The most common kind of subjects found within a permissions system are some form of **user**.
+
+Typically, it is users that are accessing your application or service and, therefore, it is for them that the various permissions must be checked.
+
+Choosing how to represent a user as a subject in Authzed/SpiceDB is very important, as the wrong choice can cause permissions checks to be incomplete or, in some cases, wrong.
+
+### Using a *stable* external identifier
+
+The most common and recommended approach for representing a user as a subject is to use a **stable** identifier for the user as the subject's object ID.
+
+For example, if the authentication system used is OIDC and provides a `sub` field, then the object IDs for the users might be the `sub` field:
+
+```
+check resource:someresource view user:goog|487306745603273
+```
+
+As the `sub` field is **guaranteed** to be stable for that particular user (if a compliant OIDC implementation), it is safe to use for checking permissions, as there is no risk that the `sub` will somehow represent a different user later.
+
+:::note
+If you have *multiple* authentication provides, then the recommendation is to define a subject type for *each* provider, to ensure a clean namespace:
+
+```zed
+definition firstprovideruser {}
+definition secondprovideruser {}
+```
+
+:::
+
+### Using a primary key
+
+The second most common approach is to have a representation of the subject in another backing data store, typically a relational database.
+
+If such a database exists, and there exists a single row per user, then using the row's primary ID (typically an integer or a UUID) represents another safe ID to use for user:
+
+```
+check resource:someresource view user:1337
```

Comment:
i think postgres can support reusing autoincre fields, so maybe warn about this?

Reply:
Only in certain types, but added a warning